quipucords/api/scanjob/tests_scanjob.py

```diff
@@ -457,11 +493,19 @@ def test_details(self, start_scan):
     def test_update_not_allowed(self, start_scan):
         """Completely update a Source."""
         data_discovery = {'sources': [self.source.id],
-                          'scan_type': ScanTask.SCAN_TYPE_CONNECT}
+                          'scan_type': ScanTask.SCAN_TYPE_CONNECT,
+                          'options': {'disable_optional_products':
```

Comment:
should have an update with not a JSON object (i.e. python dictionary) to test the validation error.